### ISSUE TITLE
React Query Migration

### ISSUE THREAD

**Issue by theodur** [department-of-veterans-affairs/va-mobile-app#6729]
# Project Status Overview

### Project Objective:

Since React Query is now installed in the mobile app repo and used for some of our endpoints, we should migrate the remaining applicable parts of the app to use it. This will mainly replace our asynchronous data fetches/updates in the Redux store. This won't entirely replace the Redux store, as we'll still need that for client state, e.g. demo mode, accessibility data, etc.
 
### Current Dependencies / Blockers
<details>
<summary>Blockers</summary>

|Blocker | Team / Owner | Est to Resolution |
|-------|---------|-------| 
|  |   |
|  |   |
|  |   |

</details>

### Monthly Update 
<details>
<summary>Updates</summary>

|Month| Project Health | Progress & Key Accomplishments| Issues/Risks/Blockers|
|-------| ---------|--------|---------|
|May 2024 | Yellow | About 2/3 of RQ work has been completed | Incomplete test data led to a number of bugs, which required us to reopen a lot of tickets
|May 2024 | Yellow | The remaining work is being handed over to the Global team | They're going to start with retesting what's already in Prod which could be risky
|June 2024 | Green | Team is regrouping on which slices should be moved over from redux to react query. These will likely be maintenance windows and settings
|6/26/24| Green| Two PRs remain which will be merged in the first sprint of August, after home screen personalization is live in Production
|7/10/24| Green| Two PRs remain which will be merged in the first sprint of August, after home screen personalization is live in Production
|7/23/24| Green| QA has started on the remaining PRs
|8/5/24| Green| QA is continuing to work on the remaining PRs. Engineering is determining if any other areas of the application could benefit from React Query in the future.
|8/20/24| Green| QA waiting on engineering to merge PRs
|9/3/24| Green| QA is working on the remaining PR. 
|9/17/24| Green| For the past two weeks, push notifications were not working in staging so the push notifications React Query slice could not be tested. QA is unblocked and testing the push notifications slice this sprint.
|10/1/24| Green| Unblocked and with ENG, since QA found a bug.
|11/7/24| Green| Final ticket is with QA, after which this epic will be closed out. 

</details>

# Detailed Project Scope 

### Problem Statement

Currently, we use Redux to manage our client and server state. Managing server state with Redux can be cumbersome because it requires us to manually track the state of our API calls, which adds complexity and is time-consuming when integrating new API endpoints in the app.

### Assumptions 

Migrating the server state in the app to use React Query will minimize complexity around managing state for async data calls.
 
### Definition of Done 
<details>
<summary>Following needs to be true</summary>

|Item | Completed | 
|-------| ---------|
| | | 
| | | 

</details>

### Project Scope
<details>
<summary>Items in and out of scope</summary>

|In Scope | Out of Scope| 
|-------| ---------|
| | | 
| | | 

</details>

### Important Project Links 
<details>
<summary>Links</summary>

- [Github](url)
- [Product Brief](url)
- [Product Canvas](url)
- Design 
   - [Discovery](url)
   - [Mockups](url)
   - [Final Design](url)
   - [Figma File](url)
- Research
   - [Document 1](url) 
   - [Document 2 ](url) 
-Analytics
   - [Document 1](url) 
</details>

### Team Members

<details>
<summary>Contact List</summary>

- **OCTO Product Leads:** Rachel Han and Ryan Thurlwell
- **PM:** Rachel Han
- **Ad Hoc PM:** Meko Hong
- **Engineering:** Jayson Perkins and Jon Bindbeutel
- **Design:** Lauren Russell
- **External Team 1 & Role:**
- **External Team 1 & Role:**

</details>

### OKR and Analytic Metrics 
<details>
<summary>Metrics</summary>

|What we are measuring | Why | Event Name 
|-------| ---------| ----- |
| | | |
| | | |

</details>

# Checklist

<details>
  <summary>Toggle Product Checklist</summary>

**Project Kickoff & Requirements Gathering**
   - [ ] Project Kickoff session
   - [ ] Initiative Scope finalized
   - [ ] Link Product Brief - [Sprint 0 ](https://docs.google.com/document/d/1WHDJ1iCt9J-acLx8_DSL4tkRKoVYqJGaRUqXuFf7ceg/edit# )
   - [ ] Link to Implementation Strategy
       - Implementation Plan
       - UAT Plan
       - Release Plan
       - Metrics Identified (what metrics are we capturing, how do they tie into stakeholder OKRs, etc)

**Pre-Production Readiness**
- [ ] Associate Readiness
  - [ ] Update product guide
  - [ ] Coordinate with contact center support teams
- [ ] What’s New Content
- [ ] App Store/In App messaging (if applicable)
- [ ] Coordinate with Release coordinator
- [ ] App store Content changes required? Y/N
- [ ] App store images needed? Y/N
- [ ] QA Signed off 
- [ ] UAT complete
- [ ] Obtain Go/No Go decision

**Feature Go Live**
- [ ] Ensure Waygate/Feature toggle release ticket submitted w/engineering & QA 
  - **Template: Turn on Waygate/Turn off feature flag**
- [ ] Coordinate with content & release coordinator if what’s new in app is required
</details>

<details>
  <summary>Toggle Team Collaboration Checklist</summary>

- [ ] Design Intent & Feasibility discussions
- [X] Research needed? No
- [ ] UAT Needed? Yes/No
- [ ] Ticket creation 
- [ ] Demo feature (mid check, end of development)
</details>

<details>
  <summary>Toggle Design/UX Checklist</summary>

- [ ] Review problem/opportunity statement
- [ ] Add questions and assumptions to product brief
- [ ] Add research links to product brief
- [ ] Document high-level research/design plans
- [ ] Create lo/hi fi wireframes & user flows if applicable
- [ ] Add links to Epic & applicable GitHub folders
- [ ] Socialize designs
- [ ] Sign off received
- [ ] Complete Design QA
- [ ] Does it require component review?
- [ ] Accessibility review needed?

</details>

<details>
  <summary>Toggle Content Checklist</summary>

- [ ] Review product brief and surface any questions, assumptions & risks
- [ ] Participate in project discovery and kickoff activities/ceremonies
- [ ] Complete comparative analysis/content research activities for net-new content
- [ ] Review current content and determine where app may need to differ and potential improvements to share with web
- [ ] Review past VA research and decisions documentation
- [ ] Support Research & UX Design in research sessions and synthesis
- [ ] Make content recommendations for lo-/hi-fi wireframes in collaboration with UX Design
- [ ] Collaborate with Sitewide Content team for alignment, improvements, and sign-off
- [ ] Participate in FE hand off
- [ ] Complete content QA
- [ ] Write copy for What's New In App
- [ ] Write copy for app stores' What's New sections.
- [ ] Write copy for updated screenshots in app stores 

</details>

<details>
  <summary>Toggle Engineering Checklist</summary>

- [ ] Are UI designs/ specs available/ready? If yes, they should be attached to  attach specs. (We are assuming the design specs have been approved)
- [ ] Are there any deadlines that need to be met?
- [ ] What is the roll out plan? (Do we need a feature toggle? is this going to be available to every user immediately or do we want to roll out to a few users first)
- [ ] Does this project require api integration? if yes, what api endpoint are we integrating with and what is the expected response body?
- [ ] Do we want to capture analytics for this projects (i.e. update GA, add logs to grafana/datadog etc)?If yes, what metrics would we like to capture?

</details>

<details>
  <summary>Toggle QA Checklist</summary>

- [ ] Create test plan
  -  **Template: QA Test Plan**
- [ ] Identify test data needs
- [ ] Gotten & incorporated test plan review
- [ ] QA Complete
- [ ] Waygate tasks complete
- [ ] Added new feature cases to the RC in TestRail
- [ ] Added feature cases to appropriate section of active cases in TestRail
- [ ] Spun up a (future) UI automation ticket for new feature
</details>

**Comment by TKDickson** [comment]
Meeting agenda for April 5 2024:

Problem statement
Fundamental approach needs changes (coordination, development, peer review, testing), due to issues uncovered.

Approach:
* Prevent further RQ from going out the door
* Assess released RQ changes (in priority order by frequency/use of feature) to find bugs in production
* Fix bugs ASAP
* Re-start RQ migration tickets not yet implemented, using a similar methodology

Relatively immediate tasks:
* Revert existing RQ changes to pre-develop 
* Assign a coordinator (can't be Therese after the April 5th meeting)
* Figure out an approach to RQ implementations (Therese has some ideas)
* Write & prioritize tickets for released RQ features work

**Comment by TKDickson** [comment]
General meeting notes
**Revert existing RQ changes to pre-develop** 
* Appointments was today/yesterday, should be easily reverted - Binny has already reverted this
* Prep high-level test plan & questions for engineers for Claims (either to use if can't revert, or basis for others if we can) - Rachael

**Assign a coordinator (can't be Therese after the April 5th meeting)**
* Binny to coordinate at least for next few days; likely to ask Dylan when he's back

This is less "everything is definitely broken" and more "we have varying levels of confidence across features, given what we've already found, so it's worth slowing down and digging in more"

**Comment by bischoffa** [comment]
Dylan confirmed all tickets have been added. I have update the labels to mark the Health team who is leading the work and also QA since QA work is needed.

**Comment by bischoffa** [comment]
@Sparowhawk @ala-yna can you update the sprints on this ticket? shows it was last worked on the 3/12 sprint but I have heard its ongoing from QAs side.

**Comment by bischoffa** [comment]
Added this epic to the SoS board given its been worked a lot in Q1 and will roll into Q2

**Comment by bischoffa** [comment]
[https://adhoc.slack.com/archives/C02N65QRRB7/p1712860981145379](https://adhoc.slack.com/archives/C02N65QRRB7/p1712860981145379)  
  
 

![](https://github.com/department-of-veterans-affairs/va-mobile-app/assets/116006847/171e5b89-a61c-41d2-b777-0110bf251962)

**Comment by ajsarkar28** [comment]
@EricaRobbins 
Updated! There's one ticket left which is with QA. It should be done by the end of the month
